references/detection/engine.py

```diff
@@ -80,13 +82,20 @@ def evaluate(model, data_loader, device):
     coco_evaluator = CocoEvaluator(coco, iou_types)
 
     for image, targets in metric_logger.log_every(data_loader, 100, header):
-        image = list(img.to(device) for img in image)
-        targets = [{k: v.to(device) for k, v in t.items()} for t in targets]
+        if move_tensors_to_device:
+            image = list(img.to(device) for img in image)
+            targets = [{k: v.to(device) for k, v in t.items()} for t in targets]
+
+        if device.type == 'cuda':
+            torch.cuda.synchronize()
 
-        torch.cuda.synchronize()
         model_time = time.time()
         outputs = model(image)
 
+        if device.type == 'xla':
+            for output in outputs:
+                torch_xla._XLAC._xla_sync_multi(list(output.values()), devices=[])
```

Comment:
does it work if we use `xm.mark_step` here? If it does not change much, better to use the api instead of the underscore functions.

Reply:
Or would the `xm.mark_step` be *after* line 99?

Reply:
yeah that would be better. less barriers = more goodness